### PR TITLE
home-assistant: upgrade to 0.104.1, add support for zwave/zha adapters, add various python modules for components

### PR DESCRIPTION
**Pull request by colemickens**
###### Motivation for this change
* Upgrade home-assistant to latest (`0.104.1`)
* Add the `home-assistant` service user to the `dialout` group so that it can access certain zwave/zha adapters
* Add the following modules:
  * `denonavr`: (denon component)
  * `braviarc-homeassistant`: (bravia component)
  * `plexauth`, `plexapi`, `plexwebsocket`: (plex component)
  * `flux_led`, `getmac`: (magichome/fluxled component)

cc: @Mic92

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by colemickens** [comment]
(the new commits were just me fixing up the maintainer for the new python packages to be myself.)

File: pkgs/development/python-modules/flux-led/default.nix [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+, aiohttp, zigpy
+, pytest }:
+
+buildPythonPackage rec {
+  pname = "flux_led";
+  version = "0.22";
+
+  nativeBuildInputs = [  ];
+  buildInputs = [  ];
+
```
Comment (jonringer) [inline pkgs/development/python-modules/flux-led/default.nix]: remove if not needed please

Comment (colemickens) [reply pkgs/development/python-modules/flux-led/default.nix]: Done.

**Review by jonringer — COMMENTED** [review]
I would really like to see some unit tests. Makes maintaining the python package set easier, especially during large package bumps.

**Comment by colemickens** [comment]
This is my first time adding a python package, I think. I assume you're meaning to enable the unit tests for these packages themselves? I will look into this, this makes sense. I was just reviewing some other PRs and saw a comment that pypi doesn't include tests and I was wondering if that meant I was not getting tests as well.

If this isn't a correct understanding, please let me know.

**Comment by jonringer** [comment]
didn't realise it was your first time dealing with python. A lot of times, packages wont package tests even when making source distributions on PyPi, so we have to check out from github or gitlab so we can build from source + test to ensure the some assumptions weren't broken outside of normal package usage.

File: pkgs/servers/home-assistant/default.nix [hunk 2]
```diff
@@ -85,7 +82,7 @@ in with py.pkgs; buildPythonApplication rec {
     owner = "home-assistant";
     repo = "home-assistant";
     rev = version;
-    sha256 = "1492q4icyhvz30fw5ysrwlnsls4iy5pv62ay3vq1ygcfnlapkqhl";
+    sha256 = "139pfcnnhi9hr6y3fyd2qaf7cmiili5alqwm4i9gpx6lwxibvyiz";
```
Comment (jonringer) [inline pkgs/servers/home-assistant/default.nix]:
```
hash mismatch in fixed-output derivation '/nix/store/q38aa7n70ci5sy5c6nv6fvszs5ss2mqj-source':
  wanted: sha256:139pfcnnhi9hr6y3fyd2qaf7cmiili5alqwm4i9gpx6lwxibvyiz
  got:    sha256:0hgf8sc6isaaxx4jbm65pl16v4qnw4wff9rw5rzd2a4dza0qr2lr
cannot build derivation '/nix/store/x33a930rbfins9dkazzh17jvi0q9m610-homeassistant-0.104.1.drv': 1 dependencies couldn't be built
error: build of '/nix/store/x33a930rbfins9dkazzh17jvi0q9m610-homeassistant-0.104.1.drv' failed
```
```suggestion
    sha256 = "0hgf8sc6isaaxx4jbm65pl16v4qnw4wff9rw5rzd2a4dza0qr2lr";
```

**Comment by colemickens** [comment]
Thanks for the attention so far, @jonringer. There are enough changes and things worthy of review that I'm going to split this PR. It was ~~greedy~~ optimistic of me to try to do all of this at once.

Replacing this one: 
* dialout: https://github.com/NixOS/nixpkgs/pull/78009
* version bump: https://github.com/NixOS/nixpkgs/pull/78010
* ha-pkgs-plex: https://github.com/NixOS/nixpkgs/pull/78011
* ha-pkgs-denon: https://github.com/NixOS/nixpkgs/pull/78012
* ha-pkgs-getmac: https://github.com/NixOS/nixpkgs/pull/78015
* ha-pkgs-fluxled: https://github.com/NixOS/nixpkgs/pull/78013
* ha-pkgs-braviarc-homeassistant: https://github.com/NixOS/nixpkgs/pull/78014

**Comment by colemickens** [comment]
Per my last reply just now, I have updated the sources for the python packages per your comments here, but I need to go back and verify that was sufficient to trigger testing, or if I need to add an attribute...

**Comment by jonringer** [comment]
sounds good, I responded to most of them

**Pull request closed without merging**